Code/Framework/AzCore/AzCore/RTTI/ReflectContext.h

```diff
@@ -601,21 +629,114 @@ namespace AZ
         }
 
-        virtual bool CanDomInvoke(const AZ::Dom::Value& arguments) const
+        bool CanDomInvoke(const AZ::Dom::Value& arguments) const override
         {
             return DomInvokeHelper<Callable>::CanInvoke(arguments);
         }
 
-        virtual AZ::Dom::Value DomInvoke(void*, const AZ::Dom::Value& arguments)
+        AZ::Dom::Value DomInvoke(const AZ::Dom::Value& arguments) override
         {
             return DomInvokeHelper<Callable>::Invoke(m_callable, arguments);
         }
 
+        AZ::Dom::Value GetAsDomValue(AZ::PointerObject instanceObject) override
+        {
+            AZ::Dom::Value result(AZ::Dom::Type::Object);
+            result[Attribute::s_typeField] = Dom::Value(Attribute::GetTypeName(), false);
+            result[Attribute::s_attributeField] = AZ::Dom::Utils::ValueFromType(this);
+
+            // For the instance field, the logic in GetPotentialClassTypeId is replicated here
+            // to determine the class type to use for the class Instance
+            if constexpr (AZStd::function_traits<Invocable>::value)
+            {
+                if constexpr (!AZStd::is_same_v<typename AZStd::function_traits<Invocable>::class_type, AZStd::Internal::error_type>)
+                {
+                    using ClassType = typename AZStd::function_traits<Invocable>::class_type;
+
+                    // Now check if the instanceObject is derived from the function traits class type using the AZ Rtti system
+                    auto rttiHelper = AZ::GetRttiHelper<ClassType>();
+                    if (void* castAddress = rttiHelper != nullptr ? rttiHelper->Cast(instanceObject.m_address, instanceObject.m_typeId) : nullptr;
+                        castAddress != nullptr)
+                    {
+                        auto classInstance = reinterpret_cast<ClassType*>(castAddress);
+                        result[Attribute::s_instanceField] = AZ::Dom::Utils::ValueFromType(classInstance);
+                    }
+                }
+                else if constexpr (AZStd::function_traits<Invocable>::arity > 0)
+                {
+                    // Grab the first argument and check if it the class type
+                    using FirstArgType = AZStd::function_traits_get_arg_t<Invocable, 0>;
+                    using CandidateClassType = AZStd::remove_pointer_t<AZStd::remove_cvref_t<FirstArgType>>;
+                    if constexpr (AZStd::is_class_v<CandidateClassType>)
+                    {
+                        // Now check if the instanceObject is derived from the first argument of the function object
+                        // If so, it will be assumed to be the instance field in the created Dom::Value
+                        auto rttiHelper = AZ::GetRttiHelper<CandidateClassType>();
+                        if (void* castAddress = rttiHelper != nullptr ? rttiHelper->Cast(instanceObject.m_address, instanceObject.m_typeId) : nullptr;
```

Comment:
Just for readability,, I recommend wrapping all occurrences of this check in parentheses or removing != nullptr

```suggestion
                        if (void* castAddress = (rttiHelper != nullptr0 ? rttiHelper->Cast(instanceObject.m_address, instanceObject.m_typeId) : nullptr;
```

Reply:
Noted, but I think the readability in this case is subjective.